tests/conftest.py

```diff
@@ -198,6 +200,50 @@ def _make_client(auth: Auth) -> RemoteSettingsClient:
     return _make_client
 
 
+@pytest.fixture(scope="session")
+def setup_client(setup_auth, make_client) -> ClientFactory:
+    return make_client(setup_auth)
+
+
+@pytest.fixture(scope="session")
+def editor_client(editor_auth, make_client) -> ClientFactory:
+    return make_client(editor_auth)
+
+
+@pytest.fixture(scope="session")
+def reviewer_client(reviewer_auth, make_client) -> ClientFactory:
+    return make_client(reviewer_auth)
+
+
+@pytest.fixture(scope="session")
+def anonymous_client(make_client) -> ClientFactory:
+    return make_client(tuple())
+
+
+@pytest.fixture(scope="session", autouse=True)
+def _setup_server(setup_client, editor_client, reviewer_client, skip_server_setup):
```

Comment:
This fixture is a little different than what we had before:

- we didn't create the editor and reviewer users by default -- it was based on what we passed to the function
- I found at least one test that didn't use `setup_server` at all, so we might not want `autouse` this
- scoping this to "session" might be fine for now, but could lead to some unexpected behavior. I'm fairly (but not 100%) certain that we'll do this stuff the first time the fixture is encountered, but not after that. So in a test, if we were to say, delete collection created by `setup_client`, and another test implicitly relied on that collection being available, we'd get an error

Reply:
It is a bit different, and maybe I was thinking about the problem from a different perspective. In my mind, this is a global setup method as most of the tests will need most of this infrastructure. 13/15 tests (12 integration, 1 browser) currently rely setup_server. Mostly the bucket, collection, and editor user account (and by proxy, the setup user account). Only a couple need the reviewer account though.

This is also why I figured autouse with a session scope was good idea. It would run once, and all of the integration tests have what they need. I hadn't considered a future test might delete a user or collection, mainly I think of these as more gate-keepers than detailed tests. But I can adjust the scope to function and/or remove autouse if you think they should all pull it when needed.

Reply:
> In my mind, this is a global setup method as most of the tests will need most of this infrastructure.

Yeah, I'd rather have a little more verbosity in tests where it's obvious what's being set up than having setup hidden in an autoused fixture

Reply:
Talking with @alexcottner on Zoom:

Since we make a lot of assumptions about what users and collections exist on a remote server:

```rst
- There should be a bucket and collection available

  - the bucket, if not specified by the ``BUCKET`` config option, should be named ``main-workspace``
  - the collection, if not specified by the ``COLLECTION`` config option, should be named ``integration-tests``

- There should be two users available

  - one user should be added to the ``editors`` group of the available collection
  - the other should be added to the ``reviewers`` group of the available collection
  - the credentials of these users should be passed in the ``EDITOR_AUTH`` and
    ``REVIEWER_AUTH`` config options respectively
```

We can make the same assumptions when consider what to set up for each test and how